File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    pure_lost = sorted(list(set(lost) - set(reserve)))
    pure_reserve = sorted(list(set(reserve) - set(lost)))
    temp_reserve = pure_reserve[:]
    answer = n - len(pure_lost)
    
    for x in pure_lost:
        if x - 1 in temp_reserve:
            answer += 1
            temp_reserve.remove(x - 1)
        elif x + 1 in temp_reserve:
            answer += 1
            temp_reserve.remove(x + 1)
            
    return answer

File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        Set<Integer> lostSet = new HashSet<>();
        Set<Integer> reserveSet = new HashSet();
        
        int answer = 0;
        
        for (int r: reserve) {
            reserveSet.add(r);
        }
        
        for (int l: lost) {
            if (reserveSet.contains(l)) reserveSet.remove(l);
            else lostSet.add(l);
        }
        
        List<Integer> sortedLost = new ArrayList<>(lostSet);
        Collections.sort(sortedLost);
        
        for (int sl: sortedLost) {
            if (reserveSet.contains(sl - 1)) {
                lostSet.remove(sl);
                reserveSet.remove(sl - 1);
            } else if (reserveSet.contains(sl + 1)) {
                lostSet.remove(sl);
                reserveSet.remove(sl + 1);
            }
        }
        
        answer = n - lostSet.size();
        
        return answer;
    }
}